### PR TITLE
Add BBEs for query-expr/query-action

### PR DESCRIPTION
**Pull request by grainier**
## Purpose
$title. This will add BBEs for following;
- Query Action
- Query Expression
- Stream from Query Expression
- String from Query Expression
- Table from Query Expression
- XML from Query Expression

### CODE REVIEW

File: examples/query-action/query_action.metatags [hunk 1]
```diff
@@ -0,0 +1,2 @@
+description: Example on using Ballerina query action.
```
Comment (Samallama) [inline examples/query-action/query_action.metatags:1]: Ideally, the description should be between 50-160 characters. This is 40 characters.

File: examples/query-action/query_action.metatags [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
+keywords: ballerina by examples, bbe, query, query action, from, join, on, where, let, do, do action, iterable, iterate
```
Comment (Samallama) [inline examples/query-action/query_action.metatags:2]: We should ideally not have more than 5-8 keywords. Shall we pick the top ones?

**Comment by grainier** [comment]
@Samallama fixed metadata. Please review & approve.

File: examples/query-action/query_action.metatags [hunk 3]
```diff
@@ -1,2 +1,2 @@
-description: Example on using Ballerina query action.
-keywords: ballerina by examples, bbe, query, query action, from, join, on, where, let, do, do action, iterable, iterate
+description: Example on performing query actions on iterable items.
+keywords: ballerina examples, query, query action, do action, iterable, iterate
```
Comment (Samallama) [inline examples/query-action/query_action.metatags:2]: Edit: keywords: ballerina by example, query, query action, do action, iterable, iterate

Comment (Samallama) [reply examples/query-action/query_action.metatags:2]: In all BBE keywords (in other files as well) change "ballerina examples" to "ballerina by example"

File: examples/query-expression-to-xml/query_expression_to_xml.metatags [hunk 4]
```diff
@@ -1,2 +1,2 @@
 description: Example on creating a xml using Ballerina query expression.
```
Comment (Samallama) [inline examples/query-expression-to-xml/query_expression_to_xml.metatags:1]: Edit: Example of creating an XML using Ballerina query expression.

File: examples/query-action/query_action.metatags [hunk 5]
```diff
@@ -1,2 +1,2 @@
-description: Example on using Ballerina query action.
-keywords: ballerina by examples, bbe, query, query action, from, join, on, where, let, do, do action, iterable, iterate
+description: Example on performing query actions on iterable items.
```
Comment (Samallama) [inline examples/query-action/query_action.metatags:1]: Edit: "Ballerina by Example on performing query actions on iterable items."

or "Example of performing query actions on iterable items."

Follow the same format for all BBEs

File: examples/query-expression-to-xml/query_expression_to_xml.metatags [hunk 6]
```diff
@@ -1,2 +1,2 @@
...
-keywords: ballerina by examples, bbe, query, query expression, xml, from, join, on, where, let, select, iterable, iterate
+keywords: ballerina examples, query, query expression, xml from query expression, iterable, iterate
```
Comment (Samallama) [inline examples/query-expression-to-xml/query_expression_to_xml.metatags:2]: Suggested edit: keywords: ballerina examples, query, query expression, xml, iterable, iterate

File: examples/query-expression-to-table/query_expression_to_table.metatags [hunk 7]
```diff
@@ -1,2 +1,2 @@
 description: Example on creating a table using Ballerina query expression.
-keywords: ballerina by examples, bbe, query, query expression, table, from, join, on, where, let, select, iterable, iterate
+keywords: ballerina examples, query, query expression, table from query expression, iterable, iterate
```
Comment (Samallama) [inline examples/query-expression-to-table/query_expression_to_table.metatags:2]: Suggested edit: keywords: ballerina by example, query, query expression, table, iterable, iterate

File: examples/query-expression-to-string/query_expression_to_string.metatags [hunk 8]
```diff
@@ -1,2 +1,2 @@
 description: Example on creating a string using Ballerina query expression.
-keywords: ballerina by examples, bbe, query, query expression, string, from, join, on, where, let, select, iterable, iterate
+keywords: ballerina examples, query, query expression, string from query expression, iterable, iterate
```
Comment (Samallama) [inline examples/query-expression-to-string/query_expression_to_string.metatags:2]: Suggested edit: keywords: ballerina by example, query, query expression, string, iterable, iterate

File: examples/query-expression-to-stream/query_expression_to_stream.metatags [hunk 9]
```diff
@@ -1,2 +1,2 @@
 description: Example on creating a stream using Ballerina query expression.
-keywords: ballerina by examples, bbe, query, query expression, stream, from, join, on, where, let, select, iterable, iterate
+keywords: ballerina examples, query, query expression, stream from query expression, iterable, iterate
```
Comment (Samallama) [inline examples/query-expression-to-stream/query_expression_to_stream.metatags:2]: Suggested edit: ballerina by example, query, query expression, stream, iterable, iterate

File: examples/query-expression/query_expression.metatags [hunk 10]
```diff
@@ -1,2 +1,2 @@
 description: Example on creating an array using Ballerina query expression.
-keywords: ballerina by examples, bbe, query, query expression, array, list, from, join, on, where, let, select, iterable, iterate
+keywords: ballerina examples, query, query expression, array from query expression, iterable, iterate
```
Comment (Samallama) [inline examples/query-expression/query_expression.metatags:2]: Suggested edit: keywords: ballerina by example, query, query expression, array, iterable, iterate

File: examples/query-action/query_action.metatags (hunk 3 above)
Comment (Samallama) [inline examples/query-action/query_action.metatags:2]:
```suggestion
keywords: ballerina by example, query, query action, do action, iterable, iterate
```

File: examples/query-expression-to-stream/query_expression_to_stream.metatags (hunk 9 above)
Comment (Samallama) [inline examples/query-expression-to-stream/query_expression_to_stream.metatags:2]:
```suggestion
keywords: ballerina by example, query, query expression, stream, iterable, iterate
```

File: examples/query-expression-to-string/query_expression_to_string.metatags (hunk 8 above)
Comment (Samallama) [inline examples/query-expression-to-string/query_expression_to_string.metatags:2]:
```suggestion
keywords: ballerina by example, query, query expression, string, iterable, iterate
```

File: examples/query-expression-to-table/query_expression_to_table.metatags (hunk 7 above)
Comment (Samallama) [inline examples/query-expression-to-table/query_expression_to_table.metatags:2]:
```suggestion
keywords: ballerina by example, query, query expression, table, iterable, iterate
```

File: examples/query-expression-to-xml/query_expression_to_xml.metatags (hunk 6 above)
Comment (Samallama) [inline examples/query-expression-to-xml/query_expression_to_xml.metatags:2]:
```suggestion
keywords: ballerina examples, query, query expression, xml, iterable, iterate
```

File: examples/query-expression/query_expression.metatags (hunk 10 above)
Comment (Samallama) [inline examples/query-expression/query_expression.metatags:2]:
```suggestion
keywords: ballerina by example, query, query expression, array, iterable, iterate
```

File: examples/query-join/query_join.metatags [hunk 11]
```diff
@@ -1,2 +1,2 @@
 description: Example on joining two iterables using Ballerina query expression.
-keywords: ballerina by examples, bbe, query, query expression, query action, join, list, from, on, where, let, select, iterable, iterate
+keywords: ballerina examples, query, query expression, join iterables, iterable, iterate
```
Comment (Samallama) [inline examples/query-join/query_join.metatags:2]:
```suggestion
keywords: ballerina by example, query, query expression, join iterables, iterable, iterate
```

File: examples/query-action/query_action.metatags (hunk 5 above)
Comment (Samallama) [inline examples/query-action/query_action.metatags:1]:
```suggestion
description: Ballerina by Example on performing query actions on iterable items.
```

File: examples/query-expression-to-stream/query_expression_to_stream.metatags [hunk 12]
```diff
@@ -1,2 +1,2 @@
 description: Example on creating a stream using Ballerina query expression.
```
Comment (Samallama) [inline examples/query-expression-to-stream/query_expression_to_stream.metatags:1]:
```suggestion
description: Ballerina by Example on creating a stream using Ballerina query expression.
```

File: examples/query-expression-to-string/query_expression_to_string.metatags [hunk 13]
```diff
@@ -1,2 +1,2 @@
 description: Example on creating a string using Ballerina query expression.
```
Comment (Samallama) [inline examples/query-expression-to-string/query_expression_to_string.metatags:1]:
```suggestion
description: Ballerina by Example on creating a string using Ballerina query expression.
```

File: examples/query-expression-to-table/query_expression_to_table.metatags [hunk 14]
```diff
@@ -1,2 +1,2 @@
 description: Example on creating a table using Ballerina query expression.
```
Comment (Samallama) [inline examples/query-expression-to-table/query_expression_to_table.metatags:1]:
```suggestion
description: Ballerina by Example on creating a table using Ballerina query expression.
```

File: examples/query-expression-to-xml/query_expression_to_xml.metatags (hunk 4 above)
Comment (Samallama) [inline examples/query-expression-to-xml/query_expression_to_xml.metatags:1]:
```suggestion
description: Ballerina by Example on creating an XML using Ballerina query expression.
```

File: examples/query-expression/query_expression.metatags [hunk 15]
```diff
@@ -1,2 +1,2 @@
 description: Example on creating an array using Ballerina query expression.
```
Comment (Samallama) [inline examples/query-expression/query_expression.metatags:1]:
```suggestion
description: Ballerina by Example on creating an array using Ballerina query expression.
```

File: examples/query-join/query_join.metatags [hunk 16]
```diff
@@ -1,2 +1,2 @@
 description: Example on joining two iterables using Ballerina query expression.
```
Comment (Samallama) [inline examples/query-join/query_join.metatags:1]:
```suggestion
description: Ballerina by Example on joining two iterables using Ballerina query expression.
```

**Comment by grainier** [comment]
@Samallama fixed metadata. Please review & approve.

File: examples/query-expression-to-stream/query_expression_to_stream.bal [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+
+    Student[] studentList = [s1, s2, s3, s4];
+
+    //The `from` clause works similarly to a `foreach` statement.
+    //It can be used to iterate any iterable value.
+    //The `reportStream` is the result of the `query` expression.
+    stream<Report> reportStream = stream from var student in studentList
+       //The `where` clause provides a way to perform conditional execution and works similarly to an `if` condition.
+       //It can refer to variables bound by the from clause.
+       //When the `where` condition evaluates to false, the iteration skips following the clauses.
```
Comment (praneesha) [inline examples/query-expression-to-stream/query_expression_to_stream.bal:31]:
```suggestion
       //When the `where` condition evaluates to false, the iteration skips the following the clauses.
```

File: examples/query-expression-to-stream/query_expression_to_stream.description [hunk 18]
```diff
@@ -0,0 +1,4 @@
+// Adding `stream` keyword as a prefix to a `query expression` allows users to create streams from query expressions.
```
Comment (praneesha) [inline examples/query-expression-to-stream/query_expression_to_stream.description:1]:
```suggestion
// Adding the `stream` keyword as a prefix to a `query expression` allows users to create streams from query expressions.
```

File: examples/query-expression-to-stream/query_expression_to_stream.description [hunk 19]
```diff
@@ -0,0 +1,4 @@
...
+// The first clause must be the `stream from` clause and must consist of a `select` clause as well.
+// The resulted `stream` is a stream that generates the emitted values from the select clause.
+// In this scenario the result of the query expression is a stream with the constraint type of the select clause.
```
Comment (praneesha) [inline examples/query-expression-to-stream/query_expression_to_stream.description:4]:
```suggestion
// In this scenario, the result of the query expression is a stream with the constraint type of the select clause.
```

File: examples/query-expression-to-stream/query_expression_to_stream.metatags [hunk 20]
```diff
@@ -0,0 +1,2 @@
+description: Ballerina By Example on creating a stream using Ballerina query expression.
```
Comment (praneesha) [inline examples/query-expression-to-stream/query_expression_to_stream.metatags:1]:
```suggestion
description: Ballerina By Example on creating a stream using a Ballerina query expression.
```

File: examples/query-expression-to-string/query_expression_to_string.bal [hunk 21]
```diff
@@ -0,0 +1,31 @@
...
+    Student s1 = {firstName: "Alex", lastName: "George", intakeYear: 2020, score: 1.5};
+    Student s2 = {firstName: "Ranjan", lastName: "Fonseka", intakeYear: 2020, score: 0.9};
+    Student s3 = {firstName: "John", lastName: "David", intakeYear: 2022, score: 1.2};
+    Student s4 = {firstName: "Gorge", lastName: "Fernando", intakeYear: 2021, score: 1.1};
+
+    Student[] studentList = [s1, s2, s3, s4];
+
+    //The `from` clause works similarly to a `foreach` statement.
+    //It can be used to iterate any iterable value.
+    //The `students` is the concatenated string of query expression results.
```
Comment (praneesha) [inline examples/query-expression-to-string/query_expression_to_string.bal:21]:
```suggestion
    //The `students` is the concatenated string of the query expression results.
```

File: examples/query-expression-to-string/query_expression_to_string.description [hunk 22]
```diff
@@ -0,0 +1,3 @@
+// Having `string` type as the target type for a `query expression` allows users to create a string from that query expression.
```
Comment (praneesha) [inline examples/query-expression-to-string/query_expression_to_string.description:1]:
```suggestion
// Having the `string` type as the target type for a `query expression` allows users to create a string from that query expression.
```

File: examples/query-expression-to-string/query_expression_to_string.description [hunk 23]
```diff
@@ -0,0 +1,3 @@
...
+// The resulted string is the concatenation of the emitted values from the select clause.
+// Therefore, every emitted value from the select clause must be of type string.
```
Comment (praneesha) [inline examples/query-expression-to-string/query_expression_to_string.description:3]:
```suggestion
// Therefore, every emitted value from the select clause must be of the type string.
```

File: examples/query-expression-to-string/query_expression_to_string.metatags [hunk 24]
```diff
@@ -0,0 +1,2 @@
+description: Ballerina By Example on creating a string using Ballerina query expression.
```
Comment (praneesha) [inline examples/query-expression-to-string/query_expression_to_string.metatags:1]:
```suggestion
description: Ballerina By Example on creating a string using a Ballerina query expression.
```

File: examples/query-expression-to-table/query_expression_to_table.bal [hunk 25]
```diff
@@ -0,0 +1,59 @@
...
+            score: newScore
+        }
+        // The `limit` clause limits the number of output items.
+        limit 2;
+
+    io:println("Student Table Information: ", studentTable);
+
+    Student[] duplicateStdList = [s1, s2, s1];
+
+    // Defines an `error` to handle key conflict.
```
Comment (praneesha) [inline examples/query-expression-to-table/query_expression_to_table.bal:48]:
```suggestion
    // Defines an `error` to handle a key conflict.
```

File: examples/query-expression-to-table/query_expression_to_table.bal [hunk 26]
```diff
@@ -0,0 +1,59 @@
...
+    error onConflictError = error("Key Conflict", message = "cannot insert student");
+
+    StudentTable|error duplicateStdTable = table key(id) from var student in duplicateStdList
+            select student
+            // `on conflict` clause gets executed when `select` emits a row
```
Comment (praneesha) [inline examples/query-expression-to-table/query_expression_to_table.bal:53]:
```suggestion
            // The `on conflict` clause gets executed when `select` emits a row
```

File: examples/query-expression-to-table/query_expression_to_table.bal [hunk 27]
```diff
@@ -0,0 +1,59 @@
...
+            // that has the same key as a row that it emitted earlier.
+            // It gives `onConflictError` error if there is a key conflict.
```
Comment (praneesha) [inline examples/query-expression-to-table/query_expression_to_table.bal:55]:
```suggestion
            // It gives an `onConflictError` error if there is a key conflict.
```

File: examples/query-expression-to-table/query_expression_to_table.description [hunk 28]
```diff
@@ -0,0 +1,5 @@
+// Adding `table` keyword with a key specifier as a prefix to a `query expression` allows users
```
Comment (praneesha) [inline examples/query-expression-to-table/query_expression_to_table.description:1]:
```suggestion
// Adding the `table` keyword with a key specifier as a prefix to a `query expression` allows users
```

File: examples/query-expression-to-table/query_expression_to_table.description [hunk 29]
```diff
@@ -0,0 +1,5 @@
...
+// to create a `table` from query expression.
```
Comment (praneesha) [inline examples/query-expression-to-table/query_expression_to_table.description:2]:
```suggestion
// to create a `table` from the query expression.
```

File: examples/query-expression-to-table/query_expression_to_table.description [hunk 30]
```diff
@@ -0,0 +1,5 @@
...
+// The first clause must be the `table key([key/s]) from` clause and must consist of a `select` clause as well.
+// Each emitted value from the `select` clause is added as a new member to construct the resultant `table`.
+// With a keyed `table`, `on conflict` clause can be used to handle key conflicts when `select` clause emits rows.
```
Comment (praneesha) [inline examples/query-expression-to-table/query_expression_to_table.description:5]:
```suggestion
// With a keyed `table`, the `on conflict` clause can be used to handle key conflicts when the `select` clause emits the rows.
```

File: examples/query-expression-to-table/query_expression_to_table.metatags [hunk 31]
```diff
@@ -0,0 +1,2 @@
+description: Ballerina By Example on creating a table using Ballerina query expression.
```
Comment (praneesha) [inline examples/query-expression-to-table/query_expression_to_table.metatags:1]:
```suggestion
description: Ballerina By Example on creating a table using a Ballerina query expression.
```

File: examples/query-expression-to-xml/query_expression_to_xml.bal [hunk 32]
```diff
@@ -0,0 +1,36 @@
...
+                        <student>
+                           <firstName>John</firstName>
+                           <lastName>David</lastName>
+                           <intakeYear>2022</intakeYear>
+                           <score>1.5</score>
+                        </student>
+                      </school>`;
+
+    // The `from` clause works similarly to a `foreach` statement.
+    // `authors` is the concatenated `xml` of `query expression` results.
```
Comment (praneesha) [inline examples/query-expression-to-xml/query_expression_to_xml.bal:27]:
```suggestion
    // The `authors` is the concatenated `xml` of the `query expression` results.
```

File: examples/query-expression-to-xml/query_expression_to_xml.bal [hunk 33]
```diff
@@ -0,0 +1,36 @@
...
+    xml authors = from var x in school/<student>/<firstName>
+        // The `select` clause is evaluated for each iteration.
+        // The emitted values values are concatenated to form the `xml` result.
```
Comment (praneesha) [inline examples/query-expression-to-xml/query_expression_to_xml.bal:30]:
```suggestion
        // The emitted values are concatenated to form the `xml` result.
```

File: examples/query-expression-to-xml/query_expression_to_xml.description [hunk 34]
```diff
@@ -0,0 +1,3 @@
+// Having `xml` type as the target type for a `query expression` allows users to create a xml from that query expression.
```
Comment (praneesha) [inline examples/query-expression-to-xml/query_expression_to_xml.description:1]:
```suggestion
// Having the `xml` type as the target type for a `query expression` allows users to create an XML from that query expression.
```

File: examples/query-expression-to-xml/query_expression_to_xml.description [hunk 35]
```diff
@@ -0,0 +1,3 @@
...
+// The resulted `xml` is the concatenation of the emitted values from the select clause.
+// Therefore, every emitted value from the select clause must be of type `xml`.
```
Comment (praneesha) [inline examples/query-expression-to-xml/query_expression_to_xml.description:3]:
```suggestion
// Therefore, every emitted value from the select clause must be of the type `xml`.
```

File: examples/query-expression-to-xml/query_expression_to_xml.metatags [hunk 36]
```diff
@@ -0,0 +1,2 @@
+description: Ballerina By Example on creating an XML using Ballerina query expression.
```
Comment (praneesha) [inline examples/query-expression-to-xml/query_expression_to_xml.metatags:1]:
```suggestion
description: Ballerina By Example on creating an XML using a Ballerina query expression.
```

File: examples/query-expression/query_expression.description [hunk 37]
```diff
@@ -1,5 +1,4 @@
 // A `query expression` is a list comprehension, which can be used with a value that is iterable with any error type.
-// It consists of four kinds of clauses: `from`, `let`, `where`, and `select`.
-// The first clause must be a `from` clause and the last clause must be a `select`.
+// It consists of following clauses: `from`, `join`, `let`, `on`, `where`, `select`, and `limit`.
```
Comment (praneesha) [inline examples/query-expression/query_expression.description:2]:
```suggestion
// It consists of the following clauses: `from`, `join`, `let`, `on`, `where`, `select`, and `limit`.
```

File: examples/query-expression/query_expression.metatags [hunk 38]
```diff
@@ -0,0 +1,2 @@
+description: Ballerina By Example on creating an array using Ballerina query expression.
```
Comment (praneesha) [inline examples/query-expression/query_expression.metatags:1]:
```suggestion
description: Ballerina By Example on creating an array using a Ballerina query expression.
```

File: examples/query-join/query_join.bal [hunk 39]
```diff
@@ -0,0 +1,59 @@
...
+
+    Student[] studentList = [s1, s2, s3, s4];
+    Department[] departmentList = [d1, d2, d3];
+
+    // The `from` clause works similarly to a `foreach` statement.
+    // It can be used to iterate any iterable value.
+    // The `reportList` is the result of the `query expression`.
+    Report[] reportList = from var student in studentList
+       // An inner equijoin is performed here.
+       // `join` iterates any iterable value similarly to `from` clause.
```
Comment (praneesha) [inline examples/query-join/query_join.bal:41]:
```suggestion
       // The `join` clause iterates any iterable value similarly to the `from` clause.
```

File: examples/query-join/query_join.bal [hunk 40]
```diff
@@ -0,0 +1,59 @@
...
+       join var department in departmentList
+       // `on` condition is used to match `student` with `department` based on `deptId`.
```
Comment (praneesha) [inline examples/query-join/query_join.bal:43]:
```suggestion
       // The `on` condition is used to match the `student` with `department` based on `deptId`.
```

File: examples/query-join/query_join.description [hunk 41]
```diff
@@ -0,0 +1,5 @@
+// `join` clause can be used to perform an inner equijoin on iterables using `query expression`.
```
Comment (praneesha) [inline examples/query-join/query_join.description:1]:
```suggestion
// The `join` clause can be used to perform an inner equijoin on iterables using the `query expression`.
```

File: examples/query-join/query_join.description [hunk 42]
```diff
@@ -0,0 +1,5 @@
...
+// It starts with `join` which behaves similarly to `from` clause, iterating any iterable value.
```
Comment (praneesha) [inline examples/query-join/query_join.description:2]:
```suggestion
// It starts with `join`, which behaves similarly to the `from` clause iterating any iterable value.
```

File: examples/query-join/query_join.description [hunk 43]
```diff
@@ -0,0 +1,5 @@
...
+// Then, an `on` condition is defined with `equals` keyword to compare the values and identify
```
Comment (praneesha) [inline examples/query-join/query_join.description:3]:
```suggestion
// Then, an `on` condition is defined with the `equals` keyword to compare the values and identify
```

File: examples/query-join/query_join.description [hunk 44]
```diff
@@ -0,0 +1,5 @@
...
+// the matching items based on the given condition. The iteration skips following the clauses
```
Comment (praneesha) [inline examples/query-join/query_join.description:4]:
```suggestion
// the matching items based on the given condition. The iteration skips the following clauses
```

File: examples/query-join/query_join.description [hunk 45]
```diff
@@ -0,0 +1,5 @@
...
+// for the non-matching items which dissatisfy the `on` condition.
```
Comment (praneesha) [inline examples/query-join/query_join.description:5]:
```suggestion
// for the non-matching items, which dissatisfy the `on` condition.
```

File: examples/query-join/query_join.metatags [hunk 46]
```diff
@@ -0,0 +1,2 @@
+description: Ballerina By Example on joining two iterables using Ballerina query expression.
```
Comment (praneesha) [inline examples/query-join/query_join.metatags:1]:
```suggestion
description: Ballerina By Example on joining two iterables using a Ballerina query expression.
```

**Pull request merged**